test/e2e/framework/pod/resource.go

```diff
@@ -205,6 +205,15 @@ func podRunning(c clientset.Interface, podName, namespace string) wait.Condition
 		}
 		switch pod.Status.Phase {
 		case v1.PodRunning:
+			// Verify that all containers are running
```

Comment:
I debate if this should be checked at this layer or if this is guaranteed by the kubelet to be true.  

/cc @kubernetes/sig-node-pr-reviews

Reply:
The documentation says that at least one container needs to be running for the `PodRunning` phase, but not all of them:
https://github.com/kubernetes/kubernetes/blob/5823cfce76780efe5897429ea59ca55f268a3123/staging/src/k8s.io/api/core/v1/types.go#L2420-L2422

So from the current implementation perspective I would say that we have to still check every container to be running or terminated.

Reply:
If 
> The documentation says that at least one container needs to be running for the PodRunning phase, but not all of them: 

is the declared state we should not be divergent from that guarantee.  

That said, I still don't think the behavior needs to be validated at this layer, unless there is a known issue.

Reply:
@saschagrunert how is this check related to the error reported in https://github.com/kubernetes/kubernetes/issues/82445 ?

Reply:
It tries to report the "are all containers running" scenario correctly, which is currently not the case. I see the point of @timothysc, so I will check if we can verify it somewhat different...